scripts/templates/changelog.ejs

```diff
@@ -55,18 +55,39 @@
         return aOrder == -1 ? bOrder == -1 ? (a || '').localeCompare(b || '') : 1 : aOrder - bOrder;
     });
 
-  for (const scope of scopes) {
-    // Do feature first, then fixes.
-    const allScopeCommits = commits.filter(x => x.scope === scope);
+  const ALLOWED_TYPES = {
+    'feat': {
+      badgeColor: 'blue',
+      title: 'Feature',
+    },
+    'fix': {
+      badgeColor: 'green',
+      title: 'Bug Fix',
+    },
+    'perf': {
+      badgeColor: 'orange',
+      title: 'Performance Improvement',
+    },
+  };
 
-    const scopeCommits = [
-      ...allScopeCommits.filter(x => x.type === 'feat'),
-      ...allScopeCommits.filter(x => x.type === 'fix'),
-    ];
+  for (const scope of scopes) {
+    const scopeCommits = commits.filter(x => x.scope === scope && !!ALLOWED_TYPES[x.type]);
 
     if (scopeCommits.length == 0) {
         continue;
     }
+
+    scopeCommits.sort((a, b) => {
+      if (a.type < b.type) {
+        return -1;
+      }
+
+      if (a.type > b.type) {
+        return 1;
+      }
+
+      return 0;
+    });
```

Comment:
```suggestion
    scopeCommits.sort((a, b) => a.type - b.type);
```

Reply:
Actually this is not correct since `a.type - b.type` will always be `NaN`

Reply:
ah.  `type` is not numeric.  Either `scopeCommits.sort((a, b) => a.localeCompare(b));` or `scopeCommits.sort((a,b) => a.type > b.type ? 1 : -1);`